tests/FacebookTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Heise\Tests\Shariff;
+
+use GuzzleHttp\Message\Request;
+use GuzzleHttp\Message\Response;
+use Heise\Shariff\Backend\Facebook;
+
+class FacebookTest extends \PHPUnit_Framework_TestCase
+{
+    public function testConfig()
+    {
+        $client = $this->getMockBuilder('GuzzleHttp\Client')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $response = $this->getMockBuilder('GuzzleHttp\Message\ResponseInterface')
+            ->getMock();
+
+        $response
+            ->method('getBody')
+            ->willReturn('access_token=tokem')
+        ;
+
+        $client
+            ->method('send')
+            ->willReturn($response)
+        ;
+
+        $client
+          ->method('createRequest')
+          ->will($this->returnCallback(array($this, 'createRequest')))
+        ;
+
+        $client->expects($this->at(0))
+            ->method('createRequest')
+            ->with(
+                'GET',
+                'https://graph.facebook.com/oauth/access_token'
+                  . '?client_id=foo&client_secret=bar&grant_type=client_credentials'
```

Comment:
OK, not sure how to break this properly, on one line it's too long and like this indentation is not correct ...